ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/ExplicitDisclosureIT.scala

```diff
@@ -319,281 +345,244 @@ final class ExplicitDisclosureIT extends LedgerTestSuite {
   test(
     "EDInconsistentDisclosedContracts",
     "The ledger rejects disclosed contracts with inconsistent metadata",
-    allocate(Parties(2)),
+    allocate(SingleParty, SingleParty),
     enabled = _.explicitDisclosure,
-  )(implicit ec => { case Participants(Participant(ledger, owner, delegate)) =>
-    for {
-      testContext <- initializeTest(
-        ledger,
-        owner,
-        delegate,
-        filterTxBy(owner, template = byTemplate, interface = None),
-      )
-
-      //      // TODO ED: Enable once the check is implemented in command interpretation
-      //      // Exercise a choice using invalid explicit disclosure (bad contract key)
-      //      errorBadKey <- testContext
-      //        .exerciseFetchDelegated(
-      //          testContext.disclosedContract
-      //            .update(_.metadata.contractKeyHash := ByteString.copyFromUtf8("BadKeyBadKeyBadKeyBadKeyBadKey00"))
-      //        )
-      //        .mustFail("using a mismatching contract key hash in metadata")
-
-      // Exercise a choice using invalid explicit disclosure (bad ledger time)
-      errorBadLet <- testContext
-        .exerciseFetchDelegated(
-          testContext.disclosedContract
-            .update(_.metadata.createdAt := com.google.protobuf.timestamp.Timestamp.of(1, 0))
+  )(implicit ec => {
+    case Participants(
+          Participant(ownerParticipant, owner),
+          Participant(delegateParticipant, delegate),
+        ) =>
+      for {
+        testContext <- initializeTest(
+          ownerParticipant = ownerParticipant,
+          delegateParticipant = delegateParticipant,
+          owner = owner,
+          delegate = delegate,
+          transactionFilter = filterTxBy(owner, template = byTemplate, interface = None),
         )
-        .mustFail("using a mismatching ledger time")
-
-      // Exercise a choice using invalid explicit disclosure (bad payload)
-      errorBadPayload <- testContext
-        .exerciseFetchDelegated(
-          testContext.disclosedContract
-            .update(
-              _.arguments := ProtoArguments.CreateArguments(
-                Delegated(delegate, testContext.contractKey).arguments
+
+        //      // TODO ED: Enable once the check is implemented in command interpretation
+        //      // Exercise a choice using invalid explicit disclosure (bad contract key)
+        //      errorBadKey <- testContext
+        //        .exerciseFetchDelegated(
+        //          testContext.disclosedContract
+        //            .update(_.metadata.contractKeyHash := ByteString.copyFromUtf8("BadKeyBadKeyBadKeyBadKeyBadKey00"))
+        //        )
+        //        .mustFail("using a mismatching contract key hash in metadata")
+
+        // Exercise a choice using invalid explicit disclosure (bad ledger time)
+        _ <- testContext
```

Comment:
this test uses an obviously invalid ledger time which may fail on simple validation, it is not clear that it would do the same with a slightly altered plausible timestamp

Reply:
Why is this obviously invalid? it actually is failing in ledger-side consistency validation (see the removed error code assertion). Do you think that this should be failing at the beginning in API command validation?

Reply:
I am looking for a stronger form of this test, where the timestamp is at a slight shift from the original. Then we can be more certain that it was the back-end ledger that rejected the request and not the API validation logic.

Reply:
Ok, I'll use a relative tstamp shift than this absolute one, good advice 👍

Reply:
8d98b50ae7e40a9cdcc8e3bf8f6d8ef4917bc975